### DISCUSSION TITLE
WLED integration

### DISCUSSION THREAD

**Discussion by kenai2011** [jasoncoon/led-mapper#10] [Ideas]
Hi Jason, 
not sure if you know about WLED? its a realy powerful project with complex features including defining a stripe as a 2D-Matrix.
I created as JSON for my project using your tool and it looks perfect in the preview but i have no idea how to convert to WLED. This is the link to their explanations: https://kno.wled.ge/advanced/mapping/

Do you think its possible to make your tool available not only for FastLED and Pixelblaze?

Thanks
Kenai

**Comment by jasoncoon** [comment]
Hey, sorry, I've never used WLED. I've read that mapping doc before, but that only seems to describe 1D mapping. I still don't understand how 2D mapping is done.

There is this issue, with an offer to implement WLED map output: https://github.com/jasoncoon/led-mapper/issues/8

**Comment by kenai2011** [comment]
Hi Jason, WLED is really mighty and  also able to define 2D Matrix.
![image](https://github.com/jasoncoon/led-mapper/assets/48085831/144dc995-656f-4ff2-80da-7bf40dbaf067)
But the definition of the pixel position is more or less manually
![image](https://github.com/jasoncoon/led-mapper/assets/48085831/c1d87c62-b83b-4de2-bb90-a1c72746a6a9)
Therefore your tool would be nice. 
Basically its only about filling the gaps with "-1" and put all values in a looong json depending on the cunt of LEDs
Given the picture above it looks like:
{"map" :[-1,2,-1,-1,-1,-1,7...   ]}
i will ask in the link you providet if this guy proceeded already
Thanks
Kenai
